spec/models/spree/address_spec.rb

```diff
@@ -0,0 +1,303 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Spree::Address, type: :model do
+  subject { Spree::Address }
+
+  context "validation" do
+    let(:country) { create :country, states_required: true }
+    let(:state) { create :state, name: 'maryland', abbr: 'md', country: country }
+    let(:address) { build(:address, country: country) }
+
+    context 'state validation' do
+      let(:state_validator) { instance_spy(Spree::Address.state_validator_class) }
+
+      it "calls the state validator" do
+        allow(Spree::Address.state_validator_class).
+          to receive(:new).with(address).
+          and_return(state_validator)
+        expect(state_validator).to receive(:perform)
+        address.valid?
+      end
+
+      # basic integration test with the validator
+      # See address/state_validator_spec for a full address state validation
+      # test suite
+      it "performs the state validation" do
+        address.country.states_required = true
+        address.state = nil
+        address.state_name = nil
+        expect(address.valid?).to eq(false)
+        expect(address.errors['state']).to eq(["can't be blank"])
+      end
+    end
+
+    it "requires phone" do
+      address.phone = ""
+      address.valid?
+      expect(address.errors["phone"]).to eq(["can't be blank"])
+    end
+
+    it "requires zipcode" do
+      address.zipcode = ""
+      address.valid?
+      expect(address.errors['zipcode']).to include("can't be blank")
+    end
+
+    it "requires name" do
+      address.name = ""
+      address.valid?
+      expect(address.errors['name']).to include("can't be blank")
+    end
```

Comment:
@DanielePalombo We decided to remove the name presence validator, no?

Reply:
Yes, but considering we'll continue to use it and that it will be in sync with first and last names, I supposed it could be a good idea to keep it and validate its presence.